webbpsf/optics.py

```diff
@@ -773,6 +796,23 @@ def __init__(self, name="unnamed BLC", kind='nircamcircular', module='A', nd_squ
         else:
             self.bar_offset = None
 
+    def get_baroff_from_siaf(self, filt, channel='LW'):
+        """ Get bar offset directly from SIAF.
+        
+        """
+        
+        siaf = pysiaf.Siaf('NIRCAM')
```

Comment:
oh good catch, that's a significant performance hit when it all adds up. 

We could go a step further, since the siaf is read and parsed when the instrument class is set up. There's a `.siaf` attribute on any `NIRCam()` class instance, which could potentially be passed in to this. But that may be out of scope for this PR. Getting it down to a single call however is definitely worth doing.